plugins/cindygl/src/js/General.js

```diff
@@ -292,6 +307,16 @@ function getPixelType() {
     else return gl.UNSIGNED_BYTE;
 }
 
+function toFloat(pixels) {
+    let res = [];
+    for (let i = 0; i < pixels.length; i++) {
+        if (can_use_texture_float) res.push(pixels[i]);
+        else if (can_use_texture_half_float) res.push(decodeFloat16(pixels[i]));
+        else res.push(pixels[i] / 255);
+    }
+    return res;
+}
+
```

Comment:
I choose the push-approach as `samples` might be a `Float32Array` or `Uint16Array`. For typed Arrays, `map` is not implemented for several browsers yet, and would even preserve the type: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/map